tools/accuracy_checker/accuracy_checker/annotation_converters/criteo_kaggle_dac.py

```diff
@@ -0,0 +1,116 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from pathlib import Path
+import numpy as np
+
+from ..representation import OAR3DTilingSegmentationAnnotation, ClassificationAnnotation
+from ..config import NumberField, StringField, PathField, BoolField
+from .format_converter import BaseFormatConverter
+from .format_converter import ConverterReturn
+
+class CriteoKaggleDACConverter(BaseFormatConverter):
+
+    __provider__ = 'criteo_kaggle_dac'
+    # annotation_types = (OAR3DTilingSegmentationAnnotation, )
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'testing_file': PathField(description="Path to testing file."),
+            "batch": NumberField(optional=True, default=128, description="Model batch"),
+            "subsample_size": NumberField(optional=True, default=0,
+                                           description="Limit total record count to batch * subsample size"),
+            "validation": BoolField(optional=True, default=True,
+                                           description="Allows to use half of dataset for validation purposes"),
+            "separator": StringField(optional=True, default='#',
+                                     description="Separator between input identifier and file identifier"),
+            "preprocessed_dir": PathField(optional=False, is_directory=True, check_exists=True,
+                                          description="Preprocessed dataset location")
+        })
+
+        return parameters
+
+    def configure(self):
+        self.src = self.get_value_from_config('testing_file')
+        self.batch = self.get_value_from_config('batch')
+        self.subsample = self.get_value_from_config('subsample_size')
+        self.validation = self.get_value_from_config('validation')
+        self.separator = self.get_value_from_config('separator')
+        self.preprocessed_dir = self.get_value_from_config('preprocessed_dir')
+
+    def convert(self, check_content=False, **kwargs):
+
+        preprocessed_folder = Path(self.preprocessed_dir)
+        input_folder = preprocessed_folder / "bs{}".format(self.batch) / 'input'
+        mask_folder = preprocessed_folder / "bs{}".format(self.batch) / 'mask'
+
+        _ = [folder.mkdir(parents=True) for folder in [input_folder, ] if not folder.exists()]
+
+        annotations = []
+
+        subfolder = 0
+        filecnt = 0
+
+        data = np.load(self.src)
+        X_int = data['X_int']
+        X_cat = data['X_cat']
+        y = data['y']
+        samples, cat_feat = X_cat.shape
+
+        samples = (samples // self.batch) * self.batch
+        start = 0
+
+        if self.subsample:
+            samples = self.subsample * self.batch if samples > self.subsample * self.batch else samples
+        elif self.validation:
```

Comment:
logically it looks strange, when user provide in config `subsample_size` and finally get in 2 times less samples that expected...
Is this key necessary?

Reply:
Ok, subsample (subsample_size) used in config only in definition_test.yml, this is analogue of annotation_converter subsample_size option. I move it into converter-specific parameters because with subsample_size resulting dataset contains random samples of source dataset, but for this model we have lS_o input with predefined data values, randomizing it leads to inference errors. In other words, we can randomize batches or input.1 and lS_i data, lS_o data must have predefined values.
"Validation" flag used only in definitions_full.yml  and defines "validation" scope of source dataset. Dataset paper says that half of test subset used for test purposes, another half for validation purposes. 
Dataset size is other 6e6 records (and much more in case of Criteo Terabyte), so ability to use only half of dataset can be useful?

Reply:
we have option `shuffle: False` for getting ordered subset